mesa/time.py

```diff
@@ -185,18 +186,16 @@ def __init__(
 
     def step(self) -> None:
         """Executes all the stages for all agents."""
-        agent_keys = self._agents.keys()
+        agent_keys = list(self._agents.keys())
         if self.shuffle:
-            agent_keys = list(agent_keys)
             self.model.random.shuffle(agent_keys)
         for stage in self.stage_list:
             for agent_key in agent_keys:
                 getattr(self._agents[agent_key], stage)()  # Run stage
             # We recompute the keys because some agents might have been removed
             # in the previous loop.
```

Comment:
Needs to document why the casting to `list` is necessary.

Reply:
done